app/server.go

```diff
@@ -92,15 +92,23 @@ func (cw *CorsWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 const TIME_TO_WAIT_FOR_CONNECTIONS_TO_CLOSE_ON_SERVER_SHUTDOWN = time.Second
 
-func redirectHTTPToHTTPS(w http.ResponseWriter, r *http.Request) {
-	if r.Host == "" {
-		http.Error(w, "Not Found", http.StatusNotFound)
+// golang.org/x/crypto/acme/autocert/autocert.go
+func handleHTTPRedirect(w http.ResponseWriter, r *http.Request) {
```

Comment:
I opted to lift these vs. use `redirectHTTPToHTTPS`, since they appear to have cleaner behaviour for `POST` requests where redirecting often doesn't have the desired semantics (namely discarding the `POST` payload in favour of an equivalent `GET` request against the endpoint).